controllers/velero.go

```diff
@@ -390,7 +390,7 @@ func (r *DPAReconciler) buildVeleroDeployment(veleroDeployment *appsv1.Deploymen
 		// see: https://github.com/vmware-tanzu/velero/blob/ed5809b7fc22f3661eeef10bdcb63f0d74472b76/pkg/install/deployment.go#L223-L261
 		// our secrets are appended to containers/volumeMounts in credentials.AppendPluginSpecificSpecs function
 		install.WithSecret(false),
-	)
+	).Spec
 	// adjust veleroDeployment from install
 	veleroDeployment.Name = deploymentName //reapply saved deploymentName and owner refs
```

Comment:
@kaovilai Isn't this redundant ? as we are only taking the `spec` values from the install pkg and the metadata remains preserved.

Reply:
We were taking the whole thing.. objectmeta fields as well.. hence the race conditions.

Reply:
I mean we do not need to reapply the name and owner refs here.

Reply:
yep.